### ISSUE TITLE
Video flickers severely on my own data

### ISSUE THREAD

**Issue by WyHy** [fudan-generative-vision/champ#41]
Hi，Thank you very much for your great work. it's really awesome！

I successfully ran the entire project on my own dataset, but the generated results seem to flicker much more severely compared to the example data. Is there a way to stabilize the results like the examples do?
The tools I used are as follows:

01. I obtained complete motion data based on the project at https://github.com/kijai/ComfyUI-champWrapper: 
1.1 DSINE Normal Map to obtain normal data 
1.2 DWpose Estimator to obtain dwpose data 
1.3 Depth Anything to obtain depth data 
1.4 DensePose Estimator to obtain semantic_map data
02. The motion data and ref image I used are attached.
Thank you very much.
![ref_image](https://github.com/fudan-generative-vision/champ/assets/13045015/c2c4b21e-9d72-492e-bc76-ab36f1827b8b)
[motion_data.zip](https://github.com/fudan-generative-vision/champ/files/14830851/motion_data.zip)

**Comment by Leoooo333** [comment]
Hi there, thanks for your detailed description. It's a great ComfyUI Wrapper for Champ. But for the `Semantic_map`, we actually use a different ColorMap and different part segmentation. This "gap" may introduce severe artifacts for diffusion models. Now we have released our SMPL & Rendering code, which can solve this problem. Let me know if it will fix the flickers👏

**Comment by WyHy** [comment]
> Hi there, thanks for your detailed description. It's a great ComfyUI Wrapper for Champ. But for the `Semantic_map`, we actually use a different ColorMap and different part segmentation. This "gap" may introduce severe artifacts for diffusion models. Now we have released our SMPL & Rendering code, which can solve this problem. Let me know if it will fix the flickers👏

Thank you very much for your response, but I couldn't find the code related to SMPL & Rendering in this project. However, I found this link in another issue: [Champ GitHub Repository](https://github.com/Leoooo333/champ). Following the instructions of this project, I completed the Fit SMPL and Transfer SMPL steps. But when I was processing Rendering with Blender, I encountered an error:

1. I executed the following command under the 4D-Humans directory:
   ```
   blender smpl_rendering.blend --background --python rendering.py --driving_path ../transfer_result/smpl_results --reference_path ../reference_imgs/images/ref.png --device 0
   ```
2. The version of Blender I used is the latest one: Blender 4.1.0 (hash 40a5e739e270 built 2024-03-26 00:31:19).
3. The error output is as follows:
   ```
   Blender 4.1.0 (hash 40a5e739e270 built 2024-03-26 00:31:19)
   Read blend: "/data/wangy/workspaces/future_dev/champv2/4D-Humans/smpl_rendering.blend"
   Traceback (most recent call last):
   File "/data/wangy/workspaces/future_dev/champv2/4D-Humans/rendering.py", line 3, in <module>
     import pandas as pd
   ModuleNotFoundError: No module named 'pandas'
   unknown argument, loading as file: --driving_path
   Error: Cannot read file "/data/wangy/workspaces/future_dev/champv2/4D-Humans/--driving_path": No such file or directory

   Blender quit
   ```
   The error message here indicates that pandas is not installed, but in fact, I already have pandas installed in my local virtual environment. Moreover, I have configured all the other parameter paths according to the documentation, but I have not been able to identify where the issue lies.
   Since I am not very familiar with Blender, could you please help me identify where the problem is?

**Comment by WyHy** [comment]
> Hi there, thanks for your detailed description. It's a great ComfyUI Wrapper for Champ. But for the `Semantic_map`, we actually use a different ColorMap and different part segmentation. This "gap" may introduce severe artifacts for diffusion models. Now we have released our SMPL & Rendering code, which can solve this problem. Let me know if it will fix the flickers👏

Additionally, I found a potential error in the file [transfer_smpl.py](https://github.com/Leoooo333/4D-Humans/blob/main/transfer_smpl.py) at line 60, where it states:
```python
smooth_smpl_path = os.path.join(driving_folder, "smpl_results", "smpls_group.npz")
```
The actual path should be:
```python
smooth_smpl_path = os.path.join(driving_folder, "smpls_group.npz")
```
After the Fit SMPL step, the `smpls_group.npz` file and the `smpl_results` folder are at the same level of hierarchy.

**Comment by lxwkobe24** [comment]
> > 谢谢！我刚刚在我的叉子/主叉中修复了它。我们使用 Blender 3.x 进行了测试。要在搅拌机中安装熊猫，[这可能](https://blender.stackexchange.com/questions/119051/how-do-i-install-pillow-and-or-pip-in-blender-python-in-windows-os)对您有所帮助！
> 
> 感谢您的及时回复。第三步“渲染”已成功执行。我将 Blender 降级到 3.5 版，并在 Blender 中安装了额外的依赖包，现在它成功运行。
> 
> 但是，我觉得它有点慢，GPU利用率也不高。还有其他加速计划或解决方案吗？是否支持多 GPU？
> 
> ![image](https://private-user-images.githubusercontent.com/13045015/320765613-72cc342e-5efd-4901-8663-59094bbdca42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2NDkzODYsIm5iZiI6MTcxMjY0OTA4NiwicGF0aCI6Ii8xMzA0NTAxNS8zMjA3NjU2MTMtNzJjYzM0MmUtNWVmZC00OTAxLTg2NjMtNTkwOTRiYmRjYTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA5VDA3NTEyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2OTMxN2U2NDcyMTg2OTUxMGQ2ZWQ0YzkxOTQxZjU3ODQ1MWFjZjYyNDRhN2RlYzk3ZmFjMDhlZGI3NWNjNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3DAHN73LgJETPk9ElmdxJljKwsmaixlz2Yz7ZVF8sDc)

please tell me how to change blender to 3.5  in linux 。thanks

**Comment by WyHy** [comment]
@lxwkobe24 
I installed Blender using the Snap tool, and the official documentation can be found at: [Installing on Linux - Blender 4.1 Manual](https://docs.blender.org/manual/en/latest/getting_started/installing/linux.html#install-from-snap). To install a specific version, you can refer to the following steps:

1. Uninstall the existing version.
2. Install the specified version using the following command:
   ```
   sudo snap install blender --channel=3.5/stable --classic
   ```

Please note that the command provided is an example and the channel name `3.5/stable` should be replaced with the desired version's channel if it's available on Snap.

**Comment by lxwkobe24** [comment]
> @lxwkobe24 I installed Blender using the Snap tool, and the official documentation can be found at: [Installing on Linux - Blender 4.1 Manual](https://docs.blender.org/manual/en/latest/getting_started/installing/linux.html#install-from-snap). To install a specific version, you can refer to the following steps:
> 
> 1. Uninstall the existing version.
> 2. Install the specified version using the following command:
>    ```
>    sudo snap install blender --channel=3.5/stable --classic
>    ```
> 
> Please note that the command provided is an example and the channel name should be replaced with the desired version's channel if it's available on Snap.`3.5/stable`

i try , look this error .
(error: cannot add authorization: open /home/AIlab/.snap/auth.json: permission denied)
how to solve this problem.thanks you reply。

**Comment by WyHy** [comment]
> > @lxwkobe24 I installed Blender using the Snap tool, and the official documentation can be found at: [Installing on Linux - Blender 4.1 Manual](https://docs.blender.org/manual/en/latest/getting_started/installing/linux.html#install-from-snap). To install a specific version, you can refer to the following steps:
> > 
> > 1. Uninstall the existing version.
> > 2. Install the specified version using the following command:
> >    ```
> >    sudo snap install blender --channel=3.5/stable --classic
> >    ```
> > 
> > Please note that the command provided is an example and the channel name should be replaced with the desired version's channel if it's available on Snap.`3.5/stable`
> 
> i try , look this error . (error: cannot add authorization: open /home/AIlab/.snap/auth.json: permission denied) how to solve this problem.thanks you reply。

This appears to be a permission error; ensure that you have sufficient privileges to access the `auth.json` file.

**Comment by lxwkobe24** [comment]
it's very strange！
 i use chmod change the permission, but i don't have this document with auth.json!
 could you tell me how to solve?
 thank you so much

**Comment by lxwkobe24** [comment]
> > > @lxwkobe24 I installed Blender using the Snap tool, and the official documentation can be found at: [Installing on Linux - Blender 4.1 Manual](https://docs.blender.org/manual/en/latest/getting_started/installing/linux.html#install-from-snap). To install a specific version, you can refer to the following steps:
> > > 
> > > 1. Uninstall the existing version.
> > > 2. Install the specified version using the following command:
> > >    ```
> > >    sudo snap install blender --channel=3.5/stable --classic
> > >    ```
> > > 
> > > Please note that the command provided is an example and the channel name should be replaced with the desired version's channel if it's available on Snap.`3.5/stable`
> > 
> > 
> > i try , look this error . (error: cannot add authorization: open /home/AIlab/.snap/auth.json: permission denied) how to solve this problem.thanks you reply。
> 
> This appears to be a permission error; ensure that you have sufficient privileges to access the file.`auth.json`

it's very strange！
i use chmod change the permission, but i don't have this document with auth.json!
could you tell me how to solve?
thank you so much

**Comment by Leoooo333** [comment]
@WyHy Yeah this blender script is a little bit slow, even with GPU. I am wondering if that's caused by slow I/O stream. We met the same problems when we rendered through the whole dataset. What's the resolution of your video? I suggest to downscale it to around 500.

**Comment by WyHy** [comment]
> @WyHy Yeah this blender script is a little bit slow, even with GPU. I am wondering if that's caused by slow I/O stream. We met the same problems when we rendered through the whole dataset. What's the resolution of your video? I suggest to downscale it to around 500.

I am using a landscape video with a resolution of 2084x1080. Once this run is finished, I will try to see if reducing the resolution can improve the processing speed. I am very much looking forward to seeing whether the use of SMPL and Blender effectively improves the flickering issue in the video. 😊

**Comment by WyHy** [comment]
> it's very strange！ i use chmod change the permission, but i don't have this document with auth.json! could you tell me how to solve? thank you so much

I apologize, but I am unable to assist with this issue as I am also new to Blender. I suggest you visit the official forums to seek a solution.

**Comment by lxwkobe24** [comment]
> > 这很奇怪！ 我使用 chmod 更改权限，但我没有这个文档auth.json！你能告诉我怎么解决吗？非常感谢
> 
> 我很抱歉，但我无法协助解决这个问题，因为我也是 Blender 的新手。我建议您访问官方论坛以寻求解决方案。
i have download blender3.5 successfull.
hello ,please tell me how to start blender .thanks

**Comment by WyHy** [comment]
@Leoooo333 
 I have completed all the processes on my own dataset. The flickering in the video is much better than before, but now two new issues have arisen:

1. I feel as though there might be a frame drop issue, as the characters appear to be jittery. I'm not sure what's causing this. I've inspected the data of the processed video frames individually, and it seems like when played in sequence, it should be very smooth, Where can this be adjusted?

2. The video clarity has decreased. It seems to be significantly different in quality compared to the frame images in the 'transfer_result' subfolder. I'm not sure what the reason for this is. Is it because my original resolution was 2084x1080 and the processing program compressed it? Where can this be adjusted?


https://github.com/fudan-generative-vision/champ/assets/13045015/ba3fbd4e-5141-4db3-bc84-19fdb7cf7c60

**Comment by Leoooo333** [comment]
> @Leoooo333 I have completed all the processes on my own dataset. The flickering in the video is much better than before, but now two new issues have arisen:
> 
> 1. I feel as though there might be a frame drop issue, as the characters appear to be jittery. I'm not sure what's causing this. I've inspected the data of the processed video frames individually, and it seems like when played in sequence, it should be very smooth, Where can this be adjusted?
> 2. The video clarity has decreased. It seems to be significantly different in quality compared to the frame images in the 'transfer_result' subfolder. I'm not sure what the reason for this is. Is it because my original resolution was 2084x1080 and the processing program compressed it? Where can this be adjusted?
> 
>  animation.23.06.36.mp4

1. It's caused by SMPL inconsistency between frames. The SMPL Smooth tutorial is coming soon!
2. Actually we will resize the input video to align with our neural network, the output will then be interpolated to the original resolution. That may cause the blur. 

For better results, I recommend using a half-body portrait with a front face as the reference image. We are going to solve the face problem(like the one shown in your full-body example) in the following version. Keep an eye on Champ🤸‍♂️

**Comment by lxwkobe24** [comment]
> 1. ```
>    sudo snap install blender --channel=3.5/stable --classic
>    ```

please tell me how to download pandas in blender ?
thanks.

**Comment by lxwkobe24** [comment]
> > > > 这很奇怪！ 我使用 chmod 更改权限，但我没有这个文档auth.json！你能告诉我怎么解决吗？非常感谢
> > > 
> > > 
> > > 我很抱歉，但我无法协助解决这个问题，因为我也是 Blender 的新手。我建议您访问官方论坛以寻求解决方案。我已经成功下载Blender3.5。您好，请告诉我如何启动Blender.thanks谢谢
> 
> 试试这个 https://github.com/Leoooo333/champ/tree/master

please tell me how to download pandas in blender ?
thanks.

**Comment by 2502128021** [comment]
> @Leoooo333 I have completed all the processes on my own dataset. The flickering in the video is much better than before, but now two new issues have arisen:
> 
>     1. I feel as though there might be a frame drop issue, as the characters appear to be jittery. I'm not sure what's causing this. I've inspected the data of the processed video frames individually, and it seems like when played in sequence, it should be very smooth, Where can this be adjusted?
> 
>     2. The video clarity has decreased. It seems to be significantly different in quality compared to the frame images in the 'transfer_result' subfolder. I'm not sure what the reason for this is. Is it because my original resolution was 2084x1080 and the processing program compressed it? Where can this be adjusted?
> 
> 
> animation.23.06.36.mp4

Hello，I‘m still confused what caused the flickers in the background? Do you know the reasons and how to solve it?

**Comment by xusongxuliangwangsulong** [comment]
![img_v3_029r_7251069a-f774-4fd6-b9b6-4cd8f8fc459g](https://github.com/fudan-generative-vision/champ/assets/165273452/264f613a-70f1-4e23-97bb-6a94e739f37f)
I used blender3.5, but I still got the same problem. @WyHy

**Comment by WyHy** [comment]
> > > > > 这很奇怪！ 我使用 chmod 更改权限，但我没有这个文档auth.json！你能告诉我怎么解决吗？非常感谢
> > > > 
> > > > 
> > > > 我很抱歉，但我无法协助解决这个问题，因为我也是 Blender 的新手。我建议您访问官方论坛以寻求解决方案。我已经成功下载Blender3.5。您好，请告诉我如何启动Blender.thanks谢谢
> > 
> > 
> > 试试这个 https://github.com/Leoooo333/champ/tree/master
> 
> please tell me how to download pandas in blender ? thanks.

My solution was to write an installation script called 'blender_python.py', the content of which is as follows:

```python
import sys
import subprocess

# Get the path to the Blender's Python executable
blender_python_executable = sys.executable

# Construct the command to install pandas using the system's pip
install_command = [blender_python_executable, "-m", "pip", "install", "tqdm"]

# Call the system's pip to install the package
subprocess.check_call(install_command)
```

Then, use the Blender command to execute this installation script:

```bash
blender --background --python blender_python.py
```

**Comment by WyHy** [comment]
> ![img_v3_029r_7251069a-f774-4fd6-b9b6-4cd8f8fc459g](https://private-user-images.githubusercontent.com/165273452/321609633-264f613a-70f1-4e23-97bb-6a94e739f37f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI4NDA2NzEsIm5iZiI6MTcxMjg0MDM3MSwicGF0aCI6Ii8xNjUyNzM0NTIvMzIxNjA5NjMzLTI2NGY2MTNhLTcwZjEtNGUyMy05N2JiLTZhOTRlNzM5ZjM3Zi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMVQxMjU5MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NTg4NTQ1MjMzNmVhNjc4MjgxNWRlNzU1YjBjNDMyYTQ2ODg3MzRhZjgyNmY5MWE5MGUzOTdlOWUxMTIzNjJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Kg5FSPriOAkO0RFX2DZ4ydHpFodhNTCMW7rOnMtAajg) I used blender3.5, but I still got the same problem. @WyHy

you need install tqdm in Blender, which operates in a separate Python environment.
follow https://github.com/fudan-generative-vision/champ/issues/41#issuecomment-2049656596

**Comment by WyHy** [comment]
> > @Leoooo333 I have completed all the processes on my own dataset. The flickering in the video is much better than before, but now two new issues have arisen:
> > ```
> > 1. I feel as though there might be a frame drop issue, as the characters appear to be jittery. I'm not sure what's causing this. I've inspected the data of the processed video frames individually, and it seems like when played in sequence, it should be very smooth, Where can this be adjusted?
> > 
> > 2. The video clarity has decreased. It seems to be significantly different in quality compared to the frame images in the 'transfer_result' subfolder. I'm not sure what the reason for this is. Is it because my original resolution was 2084x1080 and the processing program compressed it? Where can this be adjusted?
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > animation.23.06.36.mp4
> 
> Hello，I‘m still confused what caused the flickers in the background? Do you know the reasons and how to solve it?

see https://github.com/fudan-generative-vision/champ/issues/41#issuecomment-2045593731

**Comment by 2502128021** [comment]
> > > @Leoooo333 I have completed all the processes on my own dataset. The flickering in the video is much better than before, but now two new issues have arisen:
> > > ```
> > > 1. I feel as though there might be a frame drop issue, as the characters appear to be jittery. I'm not sure what's causing this. I've inspected the data of the processed video frames individually, and it seems like when played in sequence, it should be very smooth, Where can this be adjusted?
> > > 
> > > 2. The video clarity has decreased. It seems to be significantly different in quality compared to the frame images in the 'transfer_result' subfolder. I'm not sure what the reason for this is. Is it because my original resolution was 2084x1080 and the processing program compressed it? Where can this be adjusted?
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > animation.23.06.36.mp4
> > 
> > 
> > Hello，I‘m still confused what caused the flickers in the background? Do you know the reasons and how to solve it?
> 
> see [#41 (comment)](https://github.com/fudan-generative-vision/champ/issues/41#issuecomment-2045593731)

I'm confused about the flickers in the background, it is reasonable that bugs in smpl module cause the jitter of foreground(human), but the flickers in the background are also caused by this ?
